src/dcp/dcp.c

```diff
@@ -496,76 +711,68 @@ int main(int argc, char** argv)
         if (uuid_compare(src_pool_uuid, dst_pool_uuid) == 0) {
             same_pool = true;
         }
-    } 
+    }
 
     /* connect to DAOS source pool if uuid is valid */
-    if (mfu_src_file->type == DAOS) {
-        /* if DAOS source pool uuid is valid, then set source file type to DAOS */
-        daos_connect(rank, src_svc, src_pool_uuid, src_cont_uuid, &src_poh, &src_coh); 
+    if (!local_daos_error && mfu_src_file->type == DAOS) {
+        /* Open pool connection, but do not create container if non-existent */
+        rc = daos_connect(rank, src_svc, src_pool_uuid, src_cont_uuid, &src_poh, &src_coh, true, false);
+        if (rc != 0) {
+            /* rc from daos_connect is collective */
+            local_daos_error = true;
+        }
     }
 
-    if (mfu_dst_file->type == DAOS) {
-        if (!same_pool) {
-            /* if DAOS is the source and destination type, and containers are in different pools,
-             * then connect to the second pool */
-            daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &dst_poh, &dst_coh); 
+    /* If the source and destination are in the same pool,
+     * then open the container in that pool.
+     * Otherwise, connect to the second pool and open the container */
+    if (!local_daos_error && mfu_dst_file->type == DAOS) {
+        if (same_pool) {
+            /* Don't reconnect to pool, but do create container if non-existent */
+            if (rank == 0)
+                rc = daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &src_poh, &dst_coh, false, true);
+            else
+                rc = daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &src_poh, &dst_coh, false, true);
```

Comment:
Just another style thing is that we brace even one-line code blocks, again to avoid future bugs sneaking in where someone later adds a second line in the block without realizing the braces are missing.
```suggestion
            if (rank == 0) {
                rc = daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &src_poh, &dst_coh, false, true);
            } else {
                rc = daos_connect(rank, dst_svc, dst_pool_uuid, dst_cont_uuid, &src_poh, &dst_coh, false, true);
            }
```

Reply:
This one is interesting... That was a quick debug I was doing, but then removed it. There should only be a singel `daos_connect` call there, and my local branch has the correct version...
I'm investigating now to see why git ignored these changes...

Reply:
Okay this was very strange. My local repo was "up to date", but still did not match the branch on GitHub.
Anyway, I fixed it. Thanks for the catch!